recipes/master_config_post.rb

```diff
@@ -73,13 +73,12 @@
   persistent_storage node['cookbook-openshift3']['persistent_storage']
 end
 
-execute "Wait up to 30s for nodes registration [Size : #{node_servers.size.to_i}]" do
-  command "[[ `oc get node --no-headers --config=admin.kubeconfig | grep -wc \"Ready\"` -ne #{node_servers.size.to_i} ]]"
+execute "Wait up to 60s for nodes registration [Size : #{node_servers.size.to_i}]" do
+  command "[[ `oc get node --no-headers --config=admin.kubeconfig 2> /dev/null | wc -l` -eq #{node_servers.size.to_i} ]]"
```

Comment:
This change breaks provisioning on my multi-VM system; it forces me to follow the following workflow:
 1. provision master (converge will fail the first time on that line since no nodes have been provisioned yet)
 2. provision all other nodes
 3. provision master again to pass this execute resource.

Before, I could just do the following workflow:
 1. provision master
 2. provision other nodes
 3. reprovision master (to pickup the labels exported by the nodes with chef search)

Can there be a way to provision all nodes sequentially like before? Maybe just make wait until the *local* origin node is ready before continuing instead of waiting for all nodes to be registered (ready or not)?
 the provisioning of the first master fail unless I have provisioned all the other nodes at the same time.

Reply:
ok you are right :-) I did not like it either... will refactor it now. I have got a good way to solve that shit :+1: